pkg/ensurance/analyzer/sort/evict-sort.go

```diff
@@ -0,0 +1,182 @@
+
+var evictSortFunc = map[string]func(p1, p2 executor.EvictPod) int{
+	"ExtResourceEnable": evictUseExtResource,
+	"ClassAndPriority":  evictClassAndPriority,
+	"ExtResourceUsage":  evictExtCpuUsed2Limit,
+	"CpuUsage":          evictCpuUsage,
+	"RunningTime":       evictStartTime,
+}
+
+func EvictRankFuncConstruct(customize []string) EvictRankFunc {
+	var evictRankFunc EvictRankFunc
+	if len(customize) == 0 {
+		cmp := []cmpEvictFunc{}
+		for _, sortFunc := range customize {
+			if f, ok := evictSortFunc[sortFunc]; ok {
+				cmp = append(cmp,f)
+			}
+			evictRankFunc = evictOrderedBy(cmp...).Sort
+			return evictRankFunc
+		}
+	}
+	evictRankFunc = evictRankCpuPressure
+
+	return evictRankFunc
+}
+
+func evictRankCpuPressure(pods []executor.EvictPod) {
+	evictOrderedBy(evictUseExtResource, evictClassAndPriority, evictExtCpuUsed2Limit, evictCpuUsage, evictStartTime).Sort(pods)
+}
+
+// evictStartTime compares pods by pod's start time
+func evictStartTime(p1, p2 executor.EvictPod) int {
+	t1 := p1.StartTime
+	t2 := p2.StartTime
+
+	if t1 == nil && t2 == nil {
+		return 0
+	}
+	if t1 == nil {
+		// maybe t1 is just starting
+		return 1
+	}
+	if t2 == nil {
+		return -1
+	}
+
+	if t1.Before(t2) {
+		return 1
+	} else if t1.Equal(t2) {
+		return 0
+	}
+	return -1
+}
+
+// evictClassAndPriority compares pods by pod's ClassAndPriority
+func evictClassAndPriority(p1, p2 executor.EvictPod) int {
+	return int(executor.CompareClassAndPriority(p1.ClassAndPriority, p2.ClassAndPriority))
+}
+
+// evictUseExtResource compares pod by whether use ext resource
+func evictUseExtResource(p1, p2 executor.EvictPod) int {
+	use1 := utils.TransBool2Uint(p1.UseExtResource)
+	use2 := utils.TransBool2Uint(p2.UseExtResource)
+
+	if use1 == use2 {
+		return 0
+	}
+	if use1 < use2 {
+		return -1
+	}
+	return 1
+}
+
+// evictExtCpuUsed2Limit compares the partition extcpu usage of extcpu limit
+func evictExtCpuUsed2Limit(p1, p2 executor.EvictPod) int {
+	// if both pod don't use ext resource, then return
+	if p1.UseExtResource == false && p2.UseExtResource == false {
+		return 0
+	}
+
+	p1Ratio := p1.PodCPUUsage / float64(p1.ExtCpuLimit)
+	p2Ratio := p2.PodCPUUsage / float64(p2.ExtCpuLimit)
+
+	if utils.AlmostEqual(p1Ratio, p2Ratio) {
+		return 0
+	}
+	if p1Ratio < p2Ratio {
+		return -1
+	}
+	return 1
+}
+
+// evictCpuUsage compares the partition extcpu usage of extcpu limit
+func evictCpuUsage(p1, p2 executor.EvictPod) int {
+	var p1usage, p2usage float64
+	// if both pod is PodQOSBestEffort, then compare the absolute usage;otherwise, cmpare the ratio compared with PodCPUQuota
+	if p1.ClassAndPriority.PodQOSClass == v1.PodQOSBestEffort && p2.ClassAndPriority.PodQOSClass == v1.PodQOSBestEffort {
+		p1usage = p1.PodCPUUsage
+		p2usage = p2.PodCPUUsage
+	} else {
+		p1usage = p1.PodCPUUsage * p1.PodCPUPeriod / p1.PodCPUQuota
+		p2usage = p2.PodCPUUsage * p2.PodCPUPeriod / p2.PodCPUQuota
+	}
+	if utils.AlmostEqual(p1usage, p2usage) {
+		return 0
+	}
+	if p1usage < p2usage {
```

Comment:
Can we define an function to instead 

	if utils.AlmostEqual(p1Ratio, p2Ratio) {
		return 0
	}
	if p1Ratio < p2Ratio {
		return -1
	}
	return 1

and
	if utils.AlmostEqual(p1usage, p2usage) {
		return 0
	}
	if p1usage < p2usage {
		return -1
	}

Reply:
Ok